astropy/utils/tests/test_data.py

```diff
@@ -1736,14 +1736,14 @@ def read(self, length=None):
 
     monkeypatch.setattr(astropy.utils.data, "_build_urlopener", mockurl_builder)
 
+    report_length = 1024
+    real_length = 1023
     with pytest.raises(urllib.error.ContentTooShortError):
-        report_length = 1024
-        real_length = 1023
         download_file(TESTURL, cache=False)
```

Comment:
Do `report_length` and `real_length` even do anything?

Reply:
Yes, removing them breaks the test. Seems like they are used up above in MockURL class.

Reply:
It is very surprising that just setting local variables like that actually has an effect on the tests, but fixing this would go beyond the scope of this pull request.